x-pack/plugins/security_solution/public/management/pages/event_filters/view/components/modal/index.tsx

```diff
@@ -0,0 +1,147 @@
+import { Ecs } from '../../../../../../../common/ecs';
+import { EventFiltersForm } from '../form';
+import { useEventFiltersSelector } from '../../hooks';
+import {
+  getFormHasError,
+  isCreationInProgress,
+  isCreationSuccessful,
+} from '../../../store/selector';
+import { getInitialExceptionFromEvent } from '../../../store/utils';
+import { MODAL_TITLE, MODAL_SUBTITLE, ACTIONS_CONFIRM, ACTIONS_CANCEL } from './translations';
+
+import { EventFiltersNotification } from '../notification';
+
+export interface EventFiltersModalProps {
+  data: Ecs;
+  onCancel(): void;
+}
+
+const Modal = styled(EuiModal)`
+  ${({ theme }) => css`
+    width: ${theme.eui.euiBreakpoints.l};
+    max-width: ${theme.eui.euiBreakpoints.l};
+  `}
+`;
+
+const ModalHeader = styled(EuiModalHeader)`
+  flex-direction: column;
+  align-items: flex-start;
+`;
+
+const ModalHeaderSubtitle = styled.div`
+  ${({ theme }) => css`
+    color: ${theme.eui.euiColorMediumShade};
+  `}
+`;
+
+const ModalBodySection = styled.section`
+  ${({ theme }) => css`
+    padding: ${theme.eui.euiSizeS} ${theme.eui.euiSizeL};
+
+    &.builder-section {
+      overflow-y: scroll;
+    }
+  `}
+`;
+
+export const EventFiltersModal: React.FC<EventFiltersModalProps> = memo(({ data, onCancel }) => {
+  const dispatch = useDispatch<Dispatch<AppAction>>();
+  const formHasError = useEventFiltersSelector(getFormHasError);
+  const creationInProgress = useEventFiltersSelector(isCreationInProgress);
+  const creationSuccessful = useEventFiltersSelector(isCreationSuccessful);
+
+  useEffect(() => {
+    if (creationSuccessful) {
+      onCancel();
+      dispatch({
+        type: 'eventFiltersFormStateChanged',
+        payload: {
+          type: 'UninitialisedResourceState',
+        },
+      });
+    }
+  }, [creationSuccessful, onCancel, dispatch]);
+
+  // Initialize the store with the event passed as prop to allow render the form. It acts as componentDidMount
+  useEffect(() => {
+    dispatch({
+      type: 'eventFiltersInitForm',
+      payload: { entry: getInitialExceptionFromEvent(data) },
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const handleOnCancel = useCallback(() => {
+    if (creationInProgress) return;
+    onCancel();
+  }, [creationInProgress, onCancel]);
+
+  const confirmButtonMemo = useMemo(
+    () => (
+      <EuiButton
+        data-test-subj="add-exception-confirm-button"
+        fill
+        disabled={formHasError || creationInProgress}
+        onClick={() => {
+          dispatch({ type: 'eventFiltersCreateStart' });
+        }}
+        isLoading={creationInProgress}
+      >
+        {ACTIONS_CONFIRM}
+      </EuiButton>
+    ),
+    [dispatch, formHasError, creationInProgress]
+  );
+
+  const modalBodyMemo = useMemo(
+    () => (
+      <ModalBodySection className="builder-section">
+        <EventFiltersForm />
+      </ModalBodySection>
+    ),
+    []
+  );
```

Comment:
This JSX doesn't depend on the component's props. You can move it outside of the component to make it completely static and omit `useMemo` altogether.

Reply:
The idea was to ensure there is no rerenders  on the EventFiltersForm each time there is a change on some component props but it's not the case, right, thanks!